internal/handler/msg_createuser.go

```diff
@@ -29,6 +37,120 @@ func (h *Handler) MsgCreateUser(ctx context.Context, msg *wire.OpMsg) (*wire.OpM
 		return nil, lazyerrors.Error(err)
 	}
 
-	// TODO https://github.com/FerretDB/FerretDB/issues/1491
-	return nil, common.Unimplemented(document, document.Command())
+	dbName, err := common.GetRequiredParam[string](document, "$db")
+	if err != nil {
+		return nil, err
+	}
+
+	// TODO https://github.com/FerretDB/FerretDB/issues/3777
+	// TODO https://github.com/FerretDB/FerretDB/issues/3778
+	// TODO https://github.com/FerretDB/FerretDB/issues/3784
+	if dbName != "$external" && !document.Has("pwd") {
+		return nil, handlererrors.NewCommandErrorMsg(
+			handlererrors.ErrBadValue,
+			"Must provide a 'pwd' field for all user documents, except those with '$external' as the user's source db",
+		)
+	}
+
+	var username string
+	username, err = common.GetRequiredParam[string](document, document.Command())
+
+	if err != nil {
+		return nil, err
+	}
+
+	if username == "" {
+		return nil, handlererrors.NewCommandErrorMsg(
+			handlererrors.ErrBadValue,
+			"User document needs 'user' field to be non-empty",
+		)
+	}
+
+	if err = common.UnimplementedNonDefault(document, "customData", func(v any) bool {
+		if v == nil || v == types.Null {
+			return true
+		}
+
+		cd, ok := v.(*types.Document)
+		return ok && cd.Len() == 0
+	}); err != nil {
+		return nil, err
+	}
+
+	if _, err = common.GetRequiredParam[*types.Array](document, "roles"); err != nil {
+		var ce *handlererrors.CommandError
+		if errors.As(err, &ce) && ce.Code() == handlererrors.ErrBadValue {
+			return nil, handlererrors.NewCommandErrorMsg(
+				handlererrors.ErrMissingField,
+				"BSON field 'createUser.roles' is missing but a required field",
+			)
+		}
+
+		return nil, lazyerrors.Error(err)
+	}
+
+	if err = common.UnimplementedNonDefault(document, "roles", func(v any) bool {
+		r, ok := v.(*types.Array)
+		return ok && r.Len() == 0
+	}); err != nil {
+		return nil, err
+	}
+
+	if err = common.UnimplementedNonDefault(document, "digestPassword", func(v any) bool {
+		if v == nil || v == types.Null {
+			return true
+		}
+
+		dp, ok := v.(bool)
+		return ok && dp
+	}); err != nil {
+		return nil, err
+	}
+
+	// NOTE: In MongoDB, the comment field isn't saved in the database, but used for log and profiling.
```

Comment:
That's true for all `comment` parameters; they are very common in MongoDB. That comment does not add anything, in my opinion